systemd/systemd.go

```diff
@@ -576,6 +589,36 @@ func (s *systemd) getGlobalUserStatus(unitNames ...string) ([]*UnitStatus, error
 	return sts, nil
 }
 
+func (s *systemd) InactiveEnterTimestamp(unit string) (time.Time, error) {
+	// XXX: ignore stderr of systemctl command to avoid further infractions
+	//      around LP #1885597
+	out, err := s.systemctl("show", "--property", "InactiveEnterTimestamp", unit)
+	if err != nil {
+		return time.Time{}, osutil.OutputErr(out, err)
+	}
+
+	// the time returned by systemctl here will be formatted like so:
+	// InactiveEnterTimestamp=Fri 2021-04-16 15:32:21 UTC
+	// so we have to parse the time with a matching Go time format
+	splitVal := strings.SplitN(string(out), "=", 2)
+	if len(splitVal) != 2 {
+		// then we don't have an equals sign in the output, so systemctl must be
+		// broken
+		return time.Time{}, fmt.Errorf("internal error: systemctl output (%s) is malformed", string(out))
+	}
+
+	if splitVal[1] == "" {
```

Comment:
|| splitVal[1] == "0" ?

Reply:
i'll push this and some unit tests

Reply:
thanks, yes this one need unit tests

Reply:
hmm, looks like only *Monotonic=0 can happen, so that part is ok. I'll push the unit tests anyway.

Reply:
yes the non-monotonic property just returns the empty string in the case where a unit has never transitioned to inactive. FWIW, there is a unit test for this case in servicemgr_test.go, but yes we need test coverage of this function specifically, so thanks Maciej for the tests here